src/Contract/Processor/NonPhpFileProcessorInterface.php

```diff
@@ -0,0 +1,20 @@
+<?php
+declare(strict_types=1);
+
+namespace Rector\Core\Contract\Processor;
+
+use Symplify\SmartFileSystem\SmartFileInfo;
+
+interface NonPhpFileProcessorInterface
+{
+    public function process(SmartFileInfo $smartFileInfo): ?string;
+
+    public function canProcess(SmartFileInfo $smartFileInfo): bool;
+
+    public function transformOldContent(SmartFileInfo $smartFileInfo): string;
```

Comment:
What is difference to process () method?
It seems there should be just one

Reply:
The transformOldContent method is especially a workaround for the ComposerProcessor because it outputs the content differently.

Reply:
It should be merged into 1 method even in ComposerProcessor, to have clear API